application/src/main/java/org/togetherjava/tjbot/features/moderation/scam/ScamDetector.java

```diff
@@ -40,25 +41,36 @@ public boolean isScam(CharSequence message) {
     }
 
     private boolean isScam(AnalyseResults results) {
-        if (results.pingsEveryone && results.containsSuspiciousKeyword && results.hasUrl) {
+        if (results.pingsEveryone && (results.containsSuspiciousKeyword || results.hasUrl
+                || results.containsDollarSign)) {
             return true;
         }
-        return results.containsSuspiciousKeyword && results.hasSuspiciousUrl;
+
+        return Stream
+            .of(results.containsSuspiciousKeyword, results.hasSuspiciousUrl,
+                    results.containsDollarSign)
+            .filter(flag -> flag)
+            .count() >= 2;
     }
 
     private void analyzeToken(String token, AnalyseResults results) {
         if (token.isBlank()) {
             return;
         }
 
-        if (!results.pingsEveryone && "@everyone".equalsIgnoreCase(token)) {
+        if (!results.pingsEveryone
+                && ("@everyone".equalsIgnoreCase(token) || "@here".equalsIgnoreCase(token))) {
             results.pingsEveryone = true;
         }
 
         if (!results.containsSuspiciousKeyword && containsSuspiciousKeyword(token)) {
             results.containsSuspiciousKeyword = true;
         }
 
+        if (!results.containsDollarSign && token.contains("$")) {
+            results.containsDollarSign = true;
+        }
```

Comment:
Is the first check in these ifs even needed? I don't really see a benefit other than short-circuiting the checks.

Reply:
probably not. i cant remember why they have been there previously already - perhaps some sonar?
dunno but i just left it 🤷